battle-service/src/main/java/com/company/battle/services/GameService.java

```diff
@@ -1,27 +1,111 @@
 package com.company.battle.services;
 
-import com.company.common.dtos.GameDto;
+import com.company.battle.repositories.GameRepository;
+import com.company.battle.utils.GameUtils;
 import com.company.common.models.GameEntity;
+import com.company.common.models.SettingsEntity;
+import com.company.common.models.UserEntity;
+import com.company.common.models.enums.GameStatus;
+import com.company.common.models.enums.GameVisibility;
+import com.company.common.repositories.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.apache.tomcat.util.codec.binary.Base64;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
 
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
 import java.util.UUID;
 
 @Service
+@RequiredArgsConstructor
 public class GameService {
```

Comment:
there are some comments in upstream classes like controllers and dtos which will lead to changes in a service. will review after next update.